File: 876-hand-of-straights/hand-of-straights.py
class Solution(object):
    def isNStraightHand(self, hand, groupSize):
        if len(hand) % groupSize != 0:
            return False
        
        hand.sort()
        x = Counter(hand)

        for i in sorted(x.keys()):
            if x[i] > 0:
                for j in range(1, groupSize):
                    if i + j not in x or x[i + j] < x[i]:
                        return False
                    x[i + j] -= x[i]
                x[i] = 0
        
        return True

File: 876-hand-of-straights/hand-of-straights.c
int compare (const void *a, const void *b)
{
    return (*(int*)a - *(int*)b);
}

bool isNStraightHand(int* hand, int handSize, int groupSize) {
   qsort(hand, handSize, sizeof(int), compare);
   int  i, j, count;

    i = 0;
    while (1)
    {
        while (i < handSize && hand[i] == -1)
            i++;
        if (i == handSize)
            return (true);
        count = 1;
        j = i + 1;
        while (j < handSize && count < groupSize)
        {
            if (hand[j] == hand[i] + count)
            {
                hand[j] = -1;
                count++;
            }
            j++;
        }
        if (count != groupSize)
            return (false);
        i++;
    }
    return (true);
}